Plugins/Configuration/Collect-WMIObjects.ps1

```diff
@@ -0,0 +1,64 @@
+<#
+
+.SYNOPSIS
+    Plugin-Name: Collect-WMIObjects.ps1
+    
+.Description
+    This plugin collects WMI event filters, consumers, and bindigs for a remote 
+    machine.
+
+.EXAMPLE
+
+    Stand Alone Execution
+
+    .\Collect-WMIObjects.ps1 -ComputerName Test-PC
+
+    Power-Response Execution
+
+    Set ComputerName Test-PC
+    Run
+
+.NOTES
+    Author: Drew Schmitt
+    Date Created: 2/9/2019
+    Twitter: @5ynax
+    
+    Last Modified By:
+    Last Modified Date:
+    Twitter:
+  
+#>
+
+param (
+
+    [Parameter(Mandatory=$true,Position=0)]
+    [string[]]$ComputerName
+
+    )
+
+process{
+
+
+    #Loop through machines in $ComputerName to obtain data for each machine (if multiple machines are specified)
+    foreach ($Computer in $ComputerName) {
+
+        # Collect WMI Event Filters
+        $ScriptBlock_Filter = $ExecutionContext.InvokeCommand.NewScriptBlock("Get-WMIObject -Namespace root\Subscription -Class __EventFilter")
+    
+        Invoke-Command -ComputerName $Computer -ScriptBlock $ScriptBlock_Filter
+
+        # Collect WMI Event Consumers
+        $ScriptBlock_Consumer = $ExecutionContext.InvokeCommand.NewScriptBlock("Get-WMIObject -Namespace root\Subscription -Class __EventConsumer")
+    
+        Invoke-Command -ComputerName $Computer -ScriptBlock $ScriptBlock_Consumer
```

Comment:
Since these commands return different classes of PowerShell objects with different attributes, it would be best to ` | Out-File -ComputerName $Computer` them directly here so they contain their full information. If you have mismatching PowerShell objects with different properties, the first one in the list will take precedence and will be the only one to display full attributes

Reply:
@5ynax forgot to tag :)

Reply:
@valrkey do you mean Out-PRFile -Append $Computer ?

Reply:
@5ynax maybe `-Append {consumer|filter|etc}` ?

Reply:
But yes, that's what I meant :) `-ComputerName` is for chumps

Reply:
@valrkey yep, made the changes and submitted another commit to address this.